contracts/governance/Staking/Staking.sol

```diff
@@ -391,26 +388,19 @@ contract Staking is
         /// @dev Withdraw for each unlocked position.
         /// @dev Don't change FOUR_WEEKS to TWO_WEEKS, a lot of vestings already deployed with FOUR_WEEKS
         ///		workaround found, but it doesn't work with TWO_WEEKS
```

Comment:
are there are no FOUR_WEEKS in TeamVestings? else we shouldn't change FOUR_WEEKS

Reply:
@ororopickpocket as far as i concerned, there are no FOUR_WEEKS for teamVestings, but @ororopickpocket  might be the best one to confirm

Reply:
all vestings, including team vestings, are all deployed with FOUR_WEEKS.

however, we ran into issues here, because we added tokens to the existing vesting contracts. depending on when you add the tokens, they either get added to the existing lock dates, or with a 2 week difference. so, what we ended up with are several vesting contracts with unlocks every 2 weeks. 

these vesting contracts then could not withdraw the tokens, because the withdraw function on the vesting contract iterates over 4 week intervals. same for the delegation. so, james introduced a workaround on the staking contract. the _withdrawNext function was born. on each call of withdraw, it would now not only try to withdraw from the passed `until` date, but also from `until + two weeks`. same for delegation.

If we would deploy new vesting contracts which iterate over 2 weeks on the withdraw function, this would clash with the workaround. the first iteration would withdraw for the `until` date and `until + 2 weeks`. the next iteration, however, would again try to withdraw for `until + 2 weeks` and the requirement check would fail, leading to a reverted transaction. I think this was the reason for @jameshowlett977 's warning in the vesting logic contract, which got copied here by @cwsnt.

Now my reasoning why we can change it to TWO_WEEKS on the new team vesting cancellation functions:
With the new functions, the function `governanceWithdrawTokens` on the vesting contract is no longer involved. Neither are the new functions ever called from any of the vesting contracts.
`cancelTeamVesting` is completely independant of the vesting contracts, therefore we do not need to stick to the workaround, but can do it properly.

@tjcloa thanks for questioning me here and feel free to question this further :-)

Reply:
thanks for detailed explanation ororo
and thanks @cowsant for questioning ororo )
so in this case this comment is not relevant anymore and should be removed )